### PR TITLE
Week3 - HomeWork

### PR DESCRIPTION
**Pull request by Saikiruthiga**
- Demonstrating SQL injection vulnerability
 
 - [Video presentation for SQL injection](https://www.loom.com/share/96b8591aae60441cb146a2c8f9fcc0b3?sid=01e2766d-50db-4871-adfb-efb794993998)

### CODE REVIEW

File: nodejs/week3/app.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+const apiRouter = express.Router();
+app.use("/api", apiRouter);
+
+const contactsAPIRouter = express.Router();
+apiRouter.use("/contacts", contactsAPIRouter);
+
+contactsAPIRouter.get("/", async (req, res) => {
+  let query = knex.select("*").from("contacts");
+
+  // if ("sort" in req.query) {
```
Comment (babak-f) [inline nodejs/week3/app.js:17]: Let's get into the habit of not leaving commented code in the commit :)

File: nodejs/week3/app.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    const columnName = ["id", "first_name", "last_name", "email"];
+    if (
+      orderBy.length === 2 &&
+      columnName.includes(orderBy[0]) &&
+      (orderBy[1].toLowerCase() === "asc" ||
+        orderBy[1].toLowerCase() === "desc")
+    ) {
+      query = query.orderBy(orderBy[0], orderBy[1]);
+    } else {
+      return res.status(400).json({ Error: "Bad request" });
```
Comment (babak-f) [inline nodejs/week3/app.js:36]: `Error` does not match the JSON naming convention. Can you guess why? :)

**Review by babak-f — APPROVED** [review]

File: nodejs/week3/app.js (hunk 1 above)
Comment (Saikiruthiga) [reply nodejs/week3/app.js:17]: Sure, I will keep this in mind

File: nodejs/week3/app.js (hunk 2 above)
Comment (Saikiruthiga) [reply nodejs/week3/app.js:36]: It is the property name and in PascalCase

**Pull request merged**